File: frontend/src/services/backend/aiProvider.ts
// @ts-ignore
/* eslint-disable */
import { request } from '@umijs/max';

/** 获取AI供应商列表 获取所有AI供应商列表，支持分页 GET /v1/ai-providers */
export async function listAiProviders(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.ListAIProvidersParams,
  options?: { [key: string]: any },
) {
  return request<API.AIProviderSearchResponse>(`/v1/ai-providers`, {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** 创建AI供应商 创建一个新的AI供应商配置 POST /v1/ai-providers */
export async function createAiProvider(
  body: API.AIProviderRequest,
  options?: { [key: string]: any },
) {
  return request<API.Response>(`/v1/ai-providers`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 获取AI供应商详情 获取指定ID的AI供应商详细信息 GET /v1/ai-providers/${param0} */
export async function getAiProvider(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.GetAIProviderParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.AIProviderResponse>(`/v1/ai-providers/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 更新AI供应商 更新指定ID的AI供应商信息 PUT /v1/ai-providers/${param0} */
export async function updateAiProvider(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.UpdateAIProviderParams,
  body: API.AIProviderRequest,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.Response>(`/v1/ai-providers/${param0}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  });
}

/** 删除AI供应商 删除指定ID的AI供应商 DELETE /v1/ai-providers/${param0} */
export async function deleteAiProvider(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.DeleteAIProviderParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.Response>(`/v1/ai-providers/${param0}`, {
    method: 'DELETE',
    params: { ...queryParams },
    ...(options || {}),
  });
}

File: backend/internal/handler/ai_provider.go
package handler

import (
	v1 "backend/api/v1"
	"backend/internal/service"
	"net/http"
	"strconv"

	"github.com/gin-gonic/gin"
	"go.uber.org/zap"
)

type AIProviderHandler struct {
	*Handler
	aiProviderService service.AIProviderService
}

func NewAIProviderHandler(handler *Handler, aiProviderService service.AIProviderService) *AIProviderHandler {
	return &AIProviderHandler{
		Handler:           handler,
		aiProviderService: aiProviderService,
	}
}

// ListAIProviders godoc
// @Summary 获取AI供应商列表
// @Schemes
// @Description 获取所有AI供应商列表，支持分页
// @Tags AIProvider
// @Accept json
// @Produce json
// @Security Bearer
// @Param page query int true "页码"
// @Param pageSize query int true "分页大小"
// @Param name query string false "供应商名称"
// @Param providerType query string false "供应商类型"
// @Success 200 {object} v1.AIProviderSearchResponse
// @Router /v1/ai-providers [get]
// @ID ListAIProviders
func (h *AIProviderHandler) ListAIProviders(ctx *gin.Context) {
	var req v1.AIProviderSearchRequest
	if err := ctx.ShouldBind(&req); err != nil {
		h.logger.WithContext(ctx).Error("ListAIProviders bind error", zap.Error(err))
		v1.HandleError(ctx, http.StatusBadRequest, v1.ErrBadRequest, nil)
		return
	}

	data, err := h.aiProviderService.List(ctx, &req)
	if err != nil {
		h.logger.WithContext(ctx).Error("aiProviderService.List error", zap.Error(err))
		v1.HandleError(ctx, http.StatusInternalServerError, v1.ErrInternalServerError, gin.H{"error": err.Error()})
		return
	}
	v1.HandleSuccess(ctx, data)
}

// CreateAIProvider godoc
// @Summary 创建AI供应商
// @Schemes
// @Description 创建一个新的AI供应商配置
// @Tags AIProvider
// @Accept json
// @Produce json
// @Security Bearer
// @Param request body v1.AIProviderRequest true "供应商信息"
// @Success 200 {object} v1.Response
// @Router /v1/ai-providers [post]
// @ID CreateAIProvider
func (h *AIProviderHandler) CreateAIProvider(ctx *gin.Context) {
	var req v1.AIProviderRequest
	if err := ctx.ShouldBind(&req); err != nil {
		h.logger.WithContext(ctx).Error("CreateAIProvider bind error", zap.Error(err))
		v1.HandleError(ctx, http.StatusBadRequest, v1.ErrBadRequest, nil)
		return
	}

	uid := GetUserIdFromCtx(ctx)
	if err := h.aiProviderService.Create(ctx, &req, uid); err != nil {
		h.logger.WithContext(ctx).Error("aiProviderService.Create error", zap.Error(err))
		v1.HandleError(ctx, http.StatusInternalServerError, v1.ErrInternalServerError, gin.H{"error": err.Error()})
		return
	}
	v1.HandleSuccess(ctx, nil)
}

// UpdateAIProvider godoc
// @Summary 更新AI供应商
// @Schemes
// @Description 更新指定ID的AI供应商信息
// @Tags AIProvider
// @Accept json
// @Produce json
// @Security Bearer
// @Param id path uint true "供应商ID"
// @Param request body v1.AIProviderRequest true "供应商信息"
// @Success 200 {object} v1.Response
// @Router /v1/ai-providers/{id} [put]
// @ID UpdateAIProvider
func (h *AIProviderHandler) UpdateAIProvider(ctx *gin.Context) {
	idStr := ctx.Param("id")
	id, err := strconv.ParseInt(idStr, 10, 64)
	if err != nil {
		h.logger.WithContext(ctx).Error("UpdateAIProvider parse id error", zap.Error(err))
		v1.HandleError(ctx, http.StatusBadRequest, v1.ErrBadRequest, gin.H{"error": "invalid id"})
		return
	}

	var req v1.AIProviderRequest
	if err := ctx.ShouldBind(&req); err != nil {
		h.logger.WithContext(ctx).Error("UpdateAIProvider bind error", zap.Error(err))
		v1.HandleError(ctx, http.StatusBadRequest, v1.ErrBadRequest, nil)
		return
	}

	uid := GetUserIdFromCtx(ctx)
	if err := h.aiProviderService.Update(ctx, uint(id), &req, uid); err != nil {
		h.logger.WithContext(ctx).Error("aiProviderService.Update error", zap.Error(err))
		v1.HandleError(ctx, http.StatusInternalServerError, v1.ErrInternalServerError, gin.H{"error": err.Error()})
		return
	}
	v1.HandleSuccess(ctx, nil)
}

// DeleteAIProvider godoc
// @Summary 删除AI供应商
// @Schemes
// @Description 删除指定ID的AI供应商
// @Tags AIProvider
// @Accept json
// @Produce json
// @Security Bearer
// @Param id path uint true "供应商ID"
// @Success 200 {object} v1.Response
// @Router /v1/ai-providers/{id} [delete]
// @ID DeleteAIProvider
func (h *AIProviderHandler) DeleteAIProvider(ctx *gin.Context) {
	idStr := ctx.Param("id")
	id, err := strconv.ParseInt(idStr, 10, 64)
	if err != nil {
		h.logger.WithContext(ctx).Error("DeleteAIProvider parse id error", zap.Error(err))
		v1.HandleError(ctx, http.StatusBadRequest, v1.ErrBadRequest, gin.H{"error": "invalid id"})
		return
	}

	if err := h.aiProviderService.Delete(ctx, uint(id)); err != nil {
		h.logger.WithContext(ctx).Error("aiProviderService.Delete error", zap.Error(err))
		v1.HandleError(ctx, http.StatusInternalServerError, v1.ErrInternalServerError, nil)
		return
	}
	v1.HandleSuccess(ctx, nil)
}

// GetAIProvider godoc
// @Summary 获取AI供应商详情
// @Schemes
// @Description 获取指定ID的AI供应商详细信息
// @Tags AIProvider
// @Accept json
// @Produce json
// @Security Bearer
// @Param id path uint true "供应商ID"
// @Success 200 {object} v1.AIProviderResponse
// @Router /v1/ai-providers/{id} [get]
// @ID GetAIProvider
func (h *AIProviderHandler) GetAIProvider(ctx *gin.Context) {
	idStr := ctx.Param("id")
	id, err := strconv.ParseInt(idStr, 10, 64)
	if err != nil {
		h.logger.WithContext(ctx).Error("GetAIProvider parse id error", zap.Error(err))
		v1.HandleError(ctx, http.StatusBadRequest, v1.ErrBadRequest, gin.H{"error": "invalid id"})
		return
	}

	data, err := h.aiProviderService.Get(ctx, uint(id))
	if err != nil {
		h.logger.WithContext(ctx).Error("aiProviderService.Get error", zap.Error(err))
		v1.HandleError(ctx, http.StatusInternalServerError, v1.ErrInternalServerError, nil)
		return
	}
	v1.HandleSuccess(ctx, data)
}
